libs/next_gen_ui_agent/data_transform/chart/base.py

```diff
@@ -0,0 +1,159 @@
+"""Base class for chart data transformers with shared utility methods."""
+
+import logging
+from abc import abstractmethod
+from typing import Any, Generic, TypeVar
+
+from next_gen_ui_agent.data_transform import data_transformer_utils
+from next_gen_ui_agent.data_transform.data_transformer import DataTransformerBase
+from next_gen_ui_agent.data_transform.types import (
+    ChartDataPoint,
+    ChartSeries,
+    ComponentDataChart,
+    DataFieldArrayValue,
+)
+from next_gen_ui_agent.types import UIComponentMetadata
+from typing_extensions import override
+
+logger = logging.getLogger(__name__)
+
+TChart = TypeVar("TChart", bound=ComponentDataChart)
+
+
+class ChartDataTransformerBase(DataTransformerBase[TChart], Generic[TChart]):
+    """Base class for all chart transformers with shared utility methods."""
+
+    @override
+    def main_processing(self, json_data: Any, component: UIComponentMetadata):
+        """
+        Transform input data into chart format.
+
+        The transformer uses component metadata and fields to build chart data.
+
+        Args:
+            json_data: The input data to transform
+            component: Component metadata containing configuration and fields
+        """
+        logger.debug("Starting chart main_processing for %s", self.COMPONENT_NAME)
+        logger.debug("Component title: %s", component.title)
+
+        # Extract data using standard utilities (like other transformers)
+        # Charts allow nested arrays (e.g., movies[*].weeklyData[*]) because they
+        # have flattening logic in their _build_chart_data implementations
+        fields = data_transformer_utils.copy_array_fields_from_ui_component_metadata(
+            component.fields
+        )
+        data_transformer_utils.fill_fields_with_array_data(
+            fields, json_data, allow_nested_arrays=True
+        )
+
+        logger.debug("Extracted %d fields", len(fields))
+        for i, field in enumerate(fields):
```

Comment:
If I understand it correctly, this cycle is only to print debug messages. Should be behind some `if debug enabled`  not to be executed at all if debug is not enabled